src/databricks/labs/ucx/assessment/crawlers.py

```diff
@@ -80,6 +94,47 @@ def spark_version_compatibility(spark_version: str) -> str:
     return "supported"
 
 
+class AzureServicePrincipalCrawler(CrawlerBase):
+    def __init__(self, _azure_spn_list_with_data_access: list, ws: WorkspaceClient, sbe: SqlBackend, schema):
```

Comment:
why do we need this as a property? can we do without it?

Reply:
Do you mean as a class property instead of a object property? Plan is to have a single list of applications ids which can then be used in `_assess_service_principals` to check if exists.

Reply:
it makes no sense

Reply:
This is what i meant as object instance variable
```    def __init__(self, ws: WorkspaceClient, sbe: SqlBackend, schema):
        super().__init__(sbe, "hive_metastore", schema, "azure_service_principals")
        self._ws = ws
        self._azure_spn_list_with_data_access = []```

Reply:
this way is better, yes